File: promocode-issue/src/main/java/com/voshodnerd/promocode/issue/repository/PromoCodeRepository.java
package com.voshodnerd.promocode.issue.repository;

import com.voshodnerd.promocode.issue.model.PromoCode;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface PromoCodeRepository extends JpaRepository<PromoCode, Long> {
    Optional<PromoCode> findFirstByStatus(PromoCode.PromoCodeStatus status);
}

File: promocode-issue-kotlin/src/main/kotlin/com/voshodnerd/promocode/issue/repository/PromoCodeRepository.kt
package com.voshodnerd.promocode.issue.repository;

import com.voshodnerd.promocode.issue.model.PromoCode;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

interface PromoCodeRepository : JpaRepository<PromoCode, Long> {
    fun findFirstByStatus(status: PromoCode.PromoCodeStatus): PromoCode
}
